src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -391,14 +411,37 @@ public void regenerateIsland(@NotNull Island island, User user, @NotNull Schemat
     }
 
     /**
-     * Gets an {@link Island} from a location.
+     * Finds an island by the player's location with cache
      *
-     * @param location The location you are looking at
-     * @return Optional of the island at the location, empty if there is none
+     * @param player The specified Player
+     * @return An optional of the island the player is in
+     */
+    public @NotNull Optional<Island> getIslandViaPlayerLocation(Player player) {
+        if (!IridiumSkyblockAPI.getInstance().isIslandWorld(player.getWorld())) {
+            return Optional.empty();
+        }
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        if (user.getCurrentIslandVisiting() != null) {
+            if (user.getCurrentIslandVisiting().isInIsland(player.getLocation())) {
+                return Optional.of(user.getCurrentIslandVisiting());
+            }
+        }
+        Optional<Island> island = getIslandViaLocation(player.getLocation());
```

Comment:
I've tried to ignore this in all other files but I just can't anymore... :(
```suggestion
        }
        
        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
        if (user.getCurrentIslandVisiting() != null) {
            if (user.getCurrentIslandVisiting().isInIsland(player.getLocation())) {
                return Optional.of(user.getCurrentIslandVisiting());
            }
        }
        
        Optional<Island> island = getIslandViaLocation(player.getLocation());
```